Revit_Core_Engine/Convert/Environment/ToRevit/Pipe.cs

```diff
@@ -0,0 +1,152 @@
+
+            // Level
+            Level level = document.LevelBelow(Math.Min(start.Z, end.Z), settings);
+            if (level == null)
+                return null;
+
+            // Default system used for now
+            // TODO: in the future you could look for the existing connectors and check if any of them overlaps with start/end of this pipe - if so, use it in Pipe.Create.
+            // hacky/heavy way of getting all connectors in the link below - however, i would rather filter the connecting elements out by type/bounding box first for performance reasons
+            // https://thebuildingcoder.typepad.com/blog/2010/06/retrieve-mep-elements-and-connectors.html
+
+            Autodesk.Revit.DB.Plumbing.PipingSystemType pst = new FilteredElementCollector(document).OfClass(typeof(Autodesk.Revit.DB.Plumbing.PipingSystemType)).OfType<Autodesk.Revit.DB.Plumbing.PipingSystemType>().FirstOrDefault();
+
+            if (pst == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("No valid PipingSystemType can be found in the Revit model. Creating a revit Pipe requires a PipingSystemType.");
+                return null;
+            }
+
+            BH.Engine.Reflection.Compute.RecordWarning("Pipe creation will utilise the first available PipingSystemType from the Revit model.");
+
+            SectionProfile sectionProfile = pipe.SectionProperty.SectionProfile;
+            if (sectionProfile == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Pipe creation requires a SectionProfile. \n No elements created.");
+                return null;
+            }
+
+            PipeSectionProperty pipeSectionProperty = pipe.SectionProperty;
+
+            // Create Revit Pipe
+            revitPipe = Autodesk.Revit.DB.Plumbing.Pipe.Create(document, pst.Id, pipeType.Id, level.Id, start, end);
+            if (revitPipe == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("No Revit Pipe has been created. Please check inputs prior to push attempt.");
+                return null;
+            }
+
+            // Copy parameters from BHoM object to Revit element
+            revitPipe.CopyParameters(pipe, settings);
+
+            double flowRate = pipe.FlowRate;
+            revitPipe.SetParameter(BuiltInParameter.RBS_PIPE_FLOW_PARAM, flowRate);
+
+            // Get first available pipeLiningType from document
+            Autodesk.Revit.DB.Plumbing.PipeInsulationType pit = new FilteredElementCollector(document).OfClass(typeof(Autodesk.Revit.DB.Plumbing.PipeInsulationType)).FirstOrDefault() as Autodesk.Revit.DB.Plumbing.PipeInsulationType;
+            if (pit == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Any pipe insulation type needs to be present in the Revit model in order to push pipes with insulation.\n" +
+                    "Pipe has been created but no insulation has been applied.");
+            }
```

Comment:
This should be wrapped into `if (sectionProfile.InsulationProfile != null)` I think:

```
PipeInsulationType pit = null;
if (sectionProfile.InsulationProfile != null)
{
            pit = new FilteredElementCollector(document).OfClass(typeof(Autodesk.Revit.DB.Plumbing.PipeInsulationType)).FirstOrDefault() as Autodesk.Revit.DB.Plumbing.PipeInsulationType;
            if (pit == null)
            {
                BH.Engine.Reflection.Compute.RecordError("Any pipe insulation type needs to be present in the Revit model in order to push pipes with insulation.\n" +
                    "Pipe has been created but no insulation has been applied.");
            }
}
```